layers/sip.go

```diff
@@ -0,0 +1,360 @@
+// Copyright 2017 Google, Inc. All rights reserved.
```

Comment:
Did you gofmt this file ? 
All other code are passed to gofmt before being merged to master.